docs/env.yml

```diff
@@ -0,0 +1,17 @@
+name: docs-env
+channels:
+  - nodefaults
+  - conda-forge
+dependencies:
+  - breathe
+  - python=3.9
+  - r-base=4.1.2
+  - r-data.table=1.14.2
+  - r-jsonlite=1.7.2
+  - r-knitr=1.37
+  - r-matrix=1.4_0
+  - r-pkgdown=1.6.1
+  - r-rmarkdown=2.11
+  - r-roxygen2=7.1.2
```

Comment:
We don't want to pin R package versions anymore?.. https://github.com/microsoft/LightGBM/pull/2176#issuecomment-499058519

Reply:
The comment you've linked refers to a commit, https://github.com/microsoft/LightGBM/pull/2176/commits/014bee124102f4a1762e4be5316fab52e852f143, that changed the list of package from "not pinned at all" to "pin to aa specific BUILD of a specific VERSION".

This PR's changes are still pinning them to specific package VERSIONS, just not to individual BUILDS. Sometimes new builds of the same package version are pushed to conda channels, to fix issues like, for example, mistakes in metadata about declared dependencies. We should want to get such fixes automatically, as they make it more likely that the environment will solve correctly.

Reply:
> The comment you've linked refers to a commit, https://github.com/microsoft/LightGBM/commit/014bee124102f4a1762e4be5316fab52e852f143, that changed the list of package from "not pinned at all" to "pin to aa specific BUILD of a specific VERSION".

More relevant past discussion: https://github.com/microsoft/LightGBM/pull/2725#discussion_r373726849.

OK, let's not pin build numbers for now. But if RTD starts to fail, we know what to try first.